tests/health_checks/NDv4/test-bandwidth.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+##This script tests host to device bandwidth and device to host bandwidth using
+##the superbench gpu-copy benchmark. Users may pass a directory containing the
+##gpu-copy benchmar to the script. If no directory is passed a default 
+##location is used. If the gpu-copy benchmark has not installed succesfully
+##installationt can be done following these steps:
+##
+##Download the host to device and device to host test from superbench v0.5.
+#string="https://raw.githubusercontent.com/microsoft/superbenchmark/release/"\
+#	"0.5/superbench/benchmarks/micro_benchmarks/gpu_copy_performance/"\
+#	"gpu_copy.cu"
+#timeout 3m wget -q -o /dev/null $string
+#err_wget=$?
+#if [ $err_wget -ne 0 ]; then
+#	echo "wget exited with error code $err_wget"
+#	exit $err_wget;
+#fi
+#
+#
+##Compile the gpu-copy benchmark.
+#compile="nvcc -lnuma gpu_copy.cu -o gpu-copy"
+#error_nvcc="Fail: nvvc failed to compile gpu_copy.cu with error code"
+#catch_error "$compile" "$error_nvcc"
+
+
+#Catch error codes that may be thrown by the executable passed as the first
+#input, and if an error code is tripped throw the second input as a message
+catch_error() {
+	output=$($1)
+	err_code=$?
+	if [ $err_code -ne 0 ]; then
+		echo -e "\t $2 $err_code" >&2
+		exit $err_code;
+	fi
+}
+
+echo "htod and dtoh bandwidth check:"
+
+#If no arguments are passed to the script assume the gpu-copy test is located
+#in the default location. Otherwise check if it the gpu-copy test is in the 
+#location indicated by the argument. 
+if [ "$#" -eq 0 ]; then
+	CHECK_DIR="/opt/azurehpc/test/health_checks/NDv4/"
+else
+	pass=1
+	if test -f "$1/gpu-copy"; then
+		CHECK_DIR=$1
+	else
+		echo -e "\t **Fail** Directory $1 is missing the gpu-copy test."
+		exit 1;
+	fi
+fi
+#Store output from calls passed to catch_error
+output=""
+
+#Count the number of gpu-name nvidia-smi outputs.
+error_smi="**Fail** nvidia-smi failed with error code"
+#Lock graphics clocks to 1400 to eliminate any time for the GPUs to boost.
+#This likely isn't important for performance here, but we will do it anyway
```

Comment:
I think we can remove this comment. Clock frequency has a big impact on bandwidth performance.

Reply:
Yes, that's fair. Comment removed.